hash_practice/exercises.py

```diff
@@ -1,21 +1,31 @@
 
 def grouped_anagrams(strings):
-    """ This method will return an array of arrays.
-        Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
-    """
-    pass
+    anagrams = {}
+    for word in strings:
+        a = tuple(sorted(word))
+        if a in anagrams:
+            anagrams[a].append(word) 
+        else:
+            anagrams[a] = [word]
+    return list(anagrams.values())
+
 
 def top_k_frequent_elements(nums, k):
     """ This method will return the k most common elements
         In the case of a tie it will select the first occuring element.
         Time Complexity: ?
         Space Complexity: ?

```

Comment:
Looks like you're missing complexity calculations, but I haven't been dinging people for it, just calling attention to it.